### DISCUSSION TITLE
Dynamic SQL

### DISCUSSION THREAD

**Discussion by beginflints** [maxtoroq/DbExtensions#59] [Q&A] [answered]
Hello
I have some issue about dynamic sql
In this sample use type parameters are int(can null), I trying to use string and use ".Hasvalue" it doesn't work.

Thank.

![image](https://user-images.githubusercontent.com/38065497/38353198-ec190e3a-38df-11e8-8622-3040e0169732.png)

**Comment by maxtoroq** [comment]
Try `_If(value != null, ...)`

**Comment by beginflints** [comment]
Do you mean like this?
![image](https://user-images.githubusercontent.com/38065497/38463327-8dd4d6c0-3b22-11e8-9b57-f78af4b61a11.png)
it doesn't work for me, Or I do something wrong?

**Comment by beginflints** [comment]
![image](https://user-images.githubusercontent.com/38065497/38463355-1cc46ac6-3b23-11e8-812b-a5f1e378f469.png)

**Comment by maxtoroq** [comment, marked as answer]
`_If(REF_NO != null, ...)`

**Comment by beginflints** [comment]
It work. !!
It's very useful for me.
Thanks for your help!

**Comment by beginflints** [comment]
Sorry I'm found some error at runtime, 
This is my code
![image](https://user-images.githubusercontent.com/38065497/38489119-a3994c1e-3c0f-11e8-8c12-991019039ab4.png)

Variables
![image](https://user-images.githubusercontent.com/38065497/38489386-708e7a14-3c10-11e8-89c9-fa4f7ebb7c92.png)

Have some error at runtime 
![image](https://user-images.githubusercontent.com/38065497/38489052-72d72416-3c0f-11e8-9ef2-ce0f2ab9e527.png)

![image](https://user-images.githubusercontent.com/38065497/38489670-42d06d02-3c11-11e8-9877-618a51229c80.png)

![image](https://user-images.githubusercontent.com/38065497/38489697-5c7813cc-3c11-11e8-907a-642d9f9f40d2.png)

**Comment by beginflints** [comment]
I'm apologize, it's my fault, I thought is index ({0}>{1}>{2}) use sequentially only,
Now I'm used {0} for first variable each commands.
